File: src/main/java/me/tajam/jext/command/ExecutorDisc.java
package me.tajam.jext.command;

import me.tajam.jext.config.ConfigDiscManager;
import me.tajam.jext.disc.DiscContainer;
import me.tajam.jext.Log;

import org.bukkit.entity.Player;

class ExecutorDisc extends ExecutorAdapter {

  ExecutorDisc() {
    super("disc");
    addParameter(new ParameterDisc(true));
  }

  @Override
  boolean executePlayer(Player sender, String[] args) {
    final DiscContainer disc = ConfigDiscManager.getInstance().getDisc(args[0]);
    if (disc == null) {
      new Log().eror().t("Disc with the namespace ").o(args[0]).t(" doesn't exists.").send(sender);
      return true;
    }
    sender.getInventory().addItem(disc.getDiscItem());
    new Log().info().t("Obtained ").p().t(" disc.").send(sender, disc);
    return true;
  }
}

File: src/main/java/me/spartacus04/jext/commands/ExecutorDisc.kt
package me.spartacus04.jext.commands

import me.spartacus04.jext.commands.adapter.ExecutorAdapter
import me.spartacus04.jext.commands.adapter.ParameterDisc
import me.spartacus04.jext.utils.sendJEXTMessage
import org.bukkit.entity.Player

/**
 * ExecutorDisc is a class used to register the "disc" command to the plugin.
 *
 * @constructor The constructor is empty because the class does not have any properties.
 */
internal class ExecutorDisc : ExecutorAdapter("disc") {
    init {
        addParameter(ParameterDisc(true))
    }

    /**
     * The function `executePlayer` gives the sender a disc.
     *
     * @param sender The player who executed the command.
     * @param args The arguments that were passed to the command.
     */
    override fun executePlayer(sender: Player, args: Array<String>) {
        val disc = ParameterDisc.getDisc(args[0])

        if (disc == null) {
            sender.sendJEXTMessage("disc-namespace-not-found", hashMapOf(
                "namespace" to args[0]
            ))

            return
        }

        sender.inventory.addItem(disc.discItemStack)

        sender.sendJEXTMessage("disc-command-success", hashMapOf(
            "disc" to disc.displayName
        ))

        return
    }
}